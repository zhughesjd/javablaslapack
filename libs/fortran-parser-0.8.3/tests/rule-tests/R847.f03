! Test arithmetic-if-stmt
!      arithmetic-if-stmt  is  IF ( scalar-numeric-expr ) label, label, label
!
! Tested here: arithmetic-if-stmt
!
! Not tested here: scalar-numeric-expr and label.
if(x) 10, 20, 30

10 print *, "hello from label 10"
20 print *, "hello from label 20"
30 print *, "hello from label 30"

end

