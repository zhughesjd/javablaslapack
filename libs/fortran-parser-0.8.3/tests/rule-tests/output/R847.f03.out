*******************************************
args[1]: R847.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 613:part-ref: id=[@14,195:195='x',<209>,7:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 313:label: lbl=[@17,198:199='10',<10>,7:6]
R 313:label: lbl=[@20,202:203='20',<10>,7:10]
R 313:label: lbl=[@23,206:207='30',<10>,7:14]
R 847:arithmetic-if-stmt: label1=[@17,198:199='10',<10>,7:6] label2=[@20,202:203='20',<10>,7:10] label3=[@23,206:207='30',<10>,7:14]
R 214:action-stmt:
R 213:executable-construct:
R 313:label: lbl=[@26,210:211='10',<10>,9:0]
R 914:format:
R 916:output-item-list__begin:
R 427:char-literal-constant: digitString-kind-param=null identifier-kind-param=null str=[@33,222:242='"hello from label 10"',<8>,9:12]
R 306:literal-constant:
R 701:primary:
R 916:output-item:
R 916:output-item-list: count=1
R 912:print-stmt: label=[@26,210:211='10',<10>,9:0] hasOutputItemList=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 313:label: lbl=[@35,244:245='20',<10>,10:0]
R 914:format:
R 916:output-item-list__begin:
R 427:char-literal-constant: digitString-kind-param=null identifier-kind-param=null str=[@42,256:276='"hello from label 20"',<8>,10:12]
R 306:literal-constant:
R 701:primary:
R 916:output-item:
R 916:output-item-list: count=1
R 912:print-stmt: label=[@35,244:245='20',<10>,10:0] hasOutputItemList=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 313:label: lbl=[@44,278:279='30',<10>,11:0]
R 914:format:
R 916:output-item-list__begin:
R 427:char-literal-constant: digitString-kind-param=null identifier-kind-param=null str=[@51,290:310='"hello from label 30"',<8>,11:12]
R 306:literal-constant:
R 701:primary:
R 916:output-item:
R 916:output-item-list: count=1
R 912:print-stmt: label=[@44,278:279='30',<10>,11:0] hasOutputItemList=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
